refined procedures (#644)
openstack-k8s-operators/data-plane-adoption: docs_user/modules/proc_configuring-data-plane-nodes.adoc

[id="configuring-data-plane-nodes_{context}"]

= Configuring isolated networks on data plane nodes

Data plane nodes are configured by `openstack-operator` and the `OpenstackDataPlaneNodeSet` custom resources (CRs). The CRs define your desired network configuration for the nodes.

> Your {rhos_long} network configuration should reflect the existing {rhos_prev_long} ({OpenStackShort}) network setup. You must pull the `net_config.yaml` files from each {OpenStackShort} node and reuse them when you define the `OpenStackDataPlaneNodeSet` CRs. The format of the configuration does not change, so you can put network templates under `edpm_network_config_template` variables, either for all nodes or per node.

Reviewer: ```suggestion
Your {rhos_long} network configuration should reflect the existing {rhos_prev_long} ({OpenStackShort}) network setup. You must pull the `net_config.yaml` files from each {OpenStackShort} node and reuse them when you define the `OpenStackDataPlaneNodeSet` CRs. The format of the configuration does not change, so you can put network templates under `edpm_network_config_template` variables, either for all nodes or for each node.
```

> To ensure that the latest network configuration is used during the data plane adoption, you should also set `edpm_network_config_update: true` in the `nodeTemplate` attribute of the `OpenStackDataPlaneNodeSet` CR.

Reviewer: ```suggestion
To ensure that the latest network configuration is used during the data plane adoption, you should also set `edpm_network_config_update: true` in the `nodeTemplate` field of the `OpenStackDataPlaneNodeSet` CR.
```

[id="configuring-data-plane-nodes_{context}"]

= Configuring isolated networks on data plane nodes

> Data plane nodes are configured by `openstack-operator` and its CRs. The CRs define your desired network configuration for the nodes.

Author: Which CRs? Are you referring to the `OpenstackDataplaneNodeSet`CRs?

Your network configuration should reflect the existing network setup. You must pull the `net_config.yaml` files from each node and reuse them when you define the `OpenStackDataPlaneNodeSet` custom resources (CRs). The format of the configuration has not changed (`os-net-config` is still being used under the hood), so you can put network templates under `edpm_network_config_template` variables, either for all nodes or per node.

> To ensure that the latest network configuration is used during the data plane adoption, you should also set `edpm_network_config_update: true` in the `nodeTemplate` attribute of the `OpenStackDataPlaneNodeSet` CR.

Author: Can this instruction be a prerequisite to this procedure, or a step in the main procedure?

> You proceed with the data plane adoption after the {OpenShiftShort} control plane is deployed in the {OpenShiftShort} cluster. When doing so, you configure `NetConfig` and `OpenStackDataPlaneNodeSet` CRs, using the same VLAN tags and IPAM configuration as determined in the previous steps.

Author: I’m not sure I understand the significance of this paragraph. Are we referring to the data plane adoption that the customer will perform later in the adoption guide? 
I want to understand what the purpose of this procedure is so I can summarize it better. In this procedure, are customers doing the following?

1. Pull the `net_config.yaml` files from each node (from your existing RHOSP deployment?)
2. Define a `OpenStackDataPlaneNodeSet` CR (or a NetConfig CR, or both)? And the CR is supposed to include the network configuration that you want to migrate from your existing deployment to the new RHOSO deployment?
3. Is this procedure intended for the customer to prepare their nodes for the data plane adoption? That is, get the network configuration in place, then adopt the data plane?

Reviewer 2: I think it was an attempt to lay down the general path for the user, with few details. Basically, to say that you will start with control plane, then you define some CRs based on your RHOSP data plane configuration, then you proceed with data plane adoption. I don't know if we need to lay this down here. The structure of the document probably already reflects the general order and major milestones.

        start: 172.19.0.100
      cidr: 172.19.0.0/24
      vlan: 22
----

> . Optional: In the `NetConfig` CR, list multiple `allocationRanges` entries to exclude some of the IP addresses, for example, to accommodate for IP addresses already consumed by the adopted environment:

Reviewer: ```suggestion
. Optional: In the `NetConfig` CR, list multiple ranges for the `allocationRanges` field to exclude some of the IP addresses, for example, to accommodate IP addresses that are already consumed by the adopted environment:
```
